src/test/java/com/zaiats/catcher/controller/CandidateControllerTest.java

```diff
@@ -0,0 +1,107 @@
+package com.zaiats.catcher.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.zaiats.catcher.model.CandidateModel;
+import com.zaiats.catcher.service.CandidateService;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.List;
+
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ExtendWith(SpringExtension.class)
+@WebMvcTest(controllers = CandidateController.class)
+class CandidateControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @MockBean
+    private CandidateService candidateService;
+
+    private final CandidateModel candidateModel = new CandidateModel(
+            20230122222130L, "Alex", "Bip", "alex.bip@mail.com", "Developer");
+
+    @Test
+    void getAllCandidates_returns200() throws Exception {
+        when(candidateService.getAllCandidates())
+                .thenReturn(List.of(candidateModel));
+
+        mockMvc.perform(get("/candidates")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().string(
+                        objectMapper.writeValueAsString(List.of(candidateModel))))
+                .andReturn();
+
+        verify(candidateService).getAllCandidates();
+    }
+
+    @Test
+    void getCandidate_returns200() throws Exception {
+        when(candidateService.getCandidateById(20230122222130L))
+                .thenReturn(candidateModel);
+
+        mockMvc.perform(get("/candidates/{id}", 20230122222130L)
```

Comment:
1. I would like to change the ID here to `candidateModel.getId()`
2. Variable create for address

Reply:
1. Done.
2. It will be the next step after adding more tests.